launcher/src/main/java/org/apache/spark/launcher/SparkSubmitCommandBuilder.java

```diff
@@ -401,6 +403,15 @@ private boolean isThriftServer(String mainClass) {
       mainClass.equals("org.apache.spark.sql.hive.thriftserver.HiveThriftServer2"));
   }
 
+  private String findExamplesAppJar() {
+    for (String exampleJar : findExamplesJars()) {
+      if (new File(exampleJar).getName().startsWith("spark-examples")) {
+        return exampleJar;
+      }
+    }
+    throw new IllegalStateException("Failed to find examples' main app jar.");
```

Comment:
@koertkuipers I think you can fix it like:

```scala
if (isTesting) {
  SparkLauncher.NO_RESOURCE
} else {
  throw new IllegalStateException("Failed to find examples' main app jar.");
}
```

Are you interested in fixing it as a followup pr via testing as you did?